File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/consumer/config/ProductListener.java
/*
 * Copyright 2017-2019 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.configuration.kafka.docs.consumer.config;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.*;
import io.micronaut.context.annotation.Property;
import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
// end::imports[]

// tag::clazz[]
@KafkaListener(
    groupId = "products",
    pollTimeout = "500ms",
    properties = @Property(name = ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, value = "10000")
)
public class ProductListener {
// end::clazz[]

    // tag::method[]
    @Topic("awesome-products")
    public void receive(
            @KafkaKey String brand, // <1>
            Product product, // <2>
            long offset, // <3>
            int partition, // <4>
            String topic, // <5>
            long timestamp) { // <6>
        System.out.println("Got Product - " + product.getName() + " by " + brand);
    }
    // end::method[]

    // tag::consumeRecord[]
    @Topic("awesome-products")
    public void receive(ConsumerRecord<String, Product> record) { // <1>
        Product product = record.value(); // <2>
        String brand = record.key(); // <3>
        System.out.println("Got Product - " + product.getName() + " by " + brand);
    }
    // end::consumeRecord[]
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/config/ProductListener.py
import logging

# tag::imports[]
from typing import Annotated

from java.lang import Long
from micronaut.configuration.kafka.annotation import KafkaKey, KafkaListener, Topic
from micronaut.context.annotation import Property, Requires
from micronaut.kafka.docs.Product import Product
from org.apache.kafka.clients.consumer import ConsumerConfig, ConsumerRecord
# end::imports[]

LOG = logging.getLogger(__name__)


@Requires(property="spec.name", value="ConfigProductListenerTest")
# tag::clazz[]
@KafkaListener(
    groupId="products",
    pollTimeout="500ms",
    properties=Property(name=ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, value="10000")
)
class ProductListener:
# end::clazz[]

    # tag::method[]
    @Topic("awesome-products")
    def receive(self,
                brand: Annotated[str, KafkaKey],  # <1>
                product: Product,  # <2>
                offset: Long,  # <3>
                partition: int,  # <4>
                topic: str,  # <5>
                timestamp: Long) -> None:  # <6>
        LOG.info("Got Product - %s by %s", product.name, brand)
    # end::method[]

    # tag::consumeRecord[]
    @Topic("awesome-products")
    def receive_record(self, record: ConsumerRecord[str, Product]) -> None:  # <1>
        product = record.value()  # <2>
        brand = record.key()  # <3>
        LOG.info("Got Product - %s by %s", product.name, brand)
    # end::consumeRecord[]
